### PR TITLE
feat: prevent pasting of Snap name during Snap removal

### LINKED ISSUE #21445: [Bug]: Remove account snap allows copy/paste of snap name
### Describe the bug

The confirmation to remove a snap requires that the user type the snap name in.  However, the user can just copy and paste the name of the snap being removed.  This is inconsistent with the pattern established to enable eth sign, which does not allow pasting into the text box.

### Expected behavior

We have an established pattern for when the user is taking a risky action that the user must type a statement to proceed.  This can be observed with enabling eth sign under settings>advanced.  Note that when enabling eth sign the user is NOT able to copy the text and paste it into the text entry box.

For the sake of consistency we should disable copy/paste on the snap removal confirmation.

### Screenshots

_No response_

### Steps to reproduce

1. Have wallet configured and a snap account created
2. Go to settings>snaps>Snap simple Keyring
3. Scroll down and tap the CTA to `Remove MetaMask Simple Snap Keyring`
4. Tap `continue` when the accounts list is displayed
5. Copy and then paste "MetaMask Simple Snap Keyring" into the text box

### Error messages or log output

_No response_

### Version

11.4.0

### Build type

None

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### Severity

_No response_

### PR DESCRIPTION
**Pull request by montelaidev**
## **Description**

The confirmation to remove a snap that has accounts present in the keyring, requires that the user type the snap name to confirm removal. However, the user can just copy and paste the name of the snap being removed. This PR disables the ability to paste the name. 

## **Related issues**

Fixes: [21445](https://github.com/MetaMask/metamask-extension/issues/21445)

## **Manual testing steps**

1. Create a new snap account
2. Go to settings and click on snap
3. Click on the snap and then remove snap
4. Click continue on the modal
5. Try to copy and paste the snap name in the input.
6. See that the input remains empty.

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/app/snaps/keyring-snap-removal-warning/keyring-snap-removal-warning.tsx [hunk 1]
```diff
@@ -152,6 +152,10 @@ export default function KeyringRemovalSnapWarning({
                       validateConfirmationInput(e.target.value),
                     );
                   }}
+                  onPaste={(e: React.ClipboardEvent<HTMLInputElement>) => {
+                    e.preventDefault();
+                    return false;
```
Comment (FrederikBolding) [inline ui/components/app/snaps/keyring-snap-removal-warning/keyring-snap-removal-warning.tsx:157]: I don't think we need to return anything here? Aren't all event handlers void functions?

Comment (montelaidev) [reply ui/components/app/snaps/keyring-snap-removal-warning/keyring-snap-removal-warning.tsx]: Yea, you're right, this can be removed

**Review by FrederikBolding — DISMISSED** [review]

File: ui/components/app/snaps/keyring-snap-removal-warning/keyring-snap-removal-warning.test.tsx [hunk 2]
```diff
@@ -103,6 +104,26 @@ describe('Keyring Snap Remove Warning', () => {
     });
   });
 
+  it('prevents pasting of the snap name to remove the snap.', async () => {
```
Comment (danroc) [inline ui/components/app/snaps/keyring-snap-removal-warning/keyring-snap-removal-warning.test.tsx:107]:
```suggestion
  it('prevents pasting of the snap name to remove the snap', async () => {
```

**Review by FrederikBolding — APPROVED** [review]

**Review by danroc — APPROVED** [review]

**Comment by k-g-j** [comment]
Chrome Version 124.0.6367.93 manual testing steps worked as expected, and the user cannot copy and paste the snap name via the keyboard shortcut or edit menu ✅ 

https://github.com/MetaMask/metamask-extension/assets/91970214/2263be11-62d8-4bcb-a826-23cd001432d0

Firefox Version 125.0.3 manual testing steps worked as expected, and the user cannot copy and paste the snap name via the keyboard shortcut or edit menu ✅ 


https://github.com/MetaMask/metamask-extension/assets/91970214/f39ba529-2d85-412f-ae60-84b1ff254205

**Pull request merged**